File: 202-happy-number/202-happy-number.js
/**
 * @param {number} n
 * @return {boolean}
 */
var isHappy = function(n) {
    let sum = 0;
    const set = new Set();
    while(sum !== 1) {
        sum = 0;
        while(n > 0) {
            const digit = n % 10;
            sum += digit * digit;
            n = ~~(n / 10);
        }
        if(set.has(sum)) return false;
        else set.add(sum);
        n = sum;
        
    }
    return true;
};

File: 202-happy-number/202-happy-number.java
class Solution {
    public boolean isHappy(int n) {
        Set<Integer> set = new HashSet<>();
        while(n != 1) {
            int sum = 0;
            while(n > 0) {
                int num = n % 10;
                sum += num * num;
                n /= 10;
            }
            if(set.contains(sum)) return false;
            else set.add(sum);
            n = sum;
        }
        
        return true;
    }
}